syntax = "proto3";
package grpcstream;
//道路目标信息接口
service StreamService {
  //道路目标信息接口
  rpc Pubobj(PubobjRequestData) returns (ResponseData) {}
}


message ObjectExtraInfo {
  //目标物体长度(m)
  float length = 1;
  //目标物体宽度(m)
  float width = 2;
  //目标物体高度(m)
  float height = 3;
  //目标颜色，RGB 格式，使用 int32 的后三个 Byte 存储
  int32 color = 4;
  //车牌号
  string licensePlate = 5;
  //车辆品牌:0:其他(以下没有则其他) 1:丰田 2:本 田 3:日产 4:别克 5: 雪佛兰 6: 凯迪
  //拉克 7:克莱斯勒 8:道奇 9:福特 10:林 肯 11:阿斯顿马丁 12:路虎 13:莲花
  //14:名爵 15:劳斯莱斯 16:五十铃 17:雷克萨斯 18:马自达 19:三菱 20:斯巴鲁
  //21:铃木 22:雅马哈 23:奥迪 24:宝马25:奔驰 26:迈巴赫 27:欧宝 28:大众
  //29:阿尔法罗密欧 30:法拉利 31:菲亚特32:依维柯 33:兰博基尼 34:玛莎拉蒂
  //35:布加迪 36:雪特龙 37:标致 38:雷诺39:斯堪尼亚 40:奇瑞 41:东风 42:吉利
  //43:长城 44:一汽 45:金杯 46:红旗 47:比亚迪 48:蔚来 49:理想 50:小鹏 51:
  //恒大 52:捷豹 53:MINI 54:哈弗 55:长 安 56:五菱 57:荣威 58:现代 59:宝骏
  //60:广汽传祺 61:起亚 62：北汽 63:领 克 64:斯科达 65:捷达 66:英菲尼迪
  //67:启辰 68:WEY 69:江淮 69:奔腾 70:
  //斯威 71:JEEP 72:威马 73:观致 74:北京 75:星途 76:讴歌 77:东南 78:福田
  //79:中华 80:海马 81:江铃 82:大运 83:特斯拉
  int32 vehicleBrand = 6;
  //道路目标状态，int 类型的数组，比如[1, 2], 具体字段定义如下所示:
  //0: 紧急制动
  //1: 车辆失控
  //2: 逆道行驶
  repeated int32 status = 7;
}

//感知目标信息
message Object {
  //感知目标 ID
  int32 id = 1;
  //感知数据来源：
  //0：未知
  //1：摄像头
  //2：浮动车上报
  //3：毫米波雷达
  //4：激光雷达
  //5: 毫米波雷达与摄像头融合感知
  //6: 激光雷达与摄像头融合感知
  int32 source = 2;
  //目标类型:
  //0: 不明物体
  //1: 小汽车
  //2: 行人
  //3: 动物
  //4.:卡车
  //5: 警车
  //6: 落石
  //7: 救护车
  //8: 道路运维车辆
  //9: 事故目标
  //10: 道路施工
  //11: 抛撒物
  //12: 自行车
  //13: 摩托车
  //14: 公共汽车
  //15: 交通锥
  //16: 小雨
  //17: 大雨
  //18: 团雾
  int32 type = 3;
  //纬度(WGS84 坐标)
  double lat = 4;
  //经度(WGS84 坐标)
  double lng = 5;
  //海拔，单位米
  double ele = 6;
  //道路 ID，限定 7 个字符长度
  string roadId = 7;
  //行驶速度(km/h)
  float speed = 8;
  //方向角；范围为 0-360 度，正北为 0度，顺时针旋转
  float heading = 9;
  //车道编号列表，指物体所处的车道
  //ID 列表
  repeated int32 lanes = 10;
  //附加信息
  ObjectExtraInfo extraInfo = 11;
}

//道路目标信息接口上报参数
message PubobjRequestData {
  //设备 ID，全局唯一
  string deviceNo = 1;
  //上报时间戳(ms)，1970 纪元后经过 的毫秒数
  sint64 timestamp = 2;
  //感知目标列表
  repeated Object objectList = 3;
}


message ResponseData {
  int32 code = 1;
  string msg = 2;
}
